File: Client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    pageNumber: number = 1;
    pageSize: number = 6;
    gender: string;
    minAge: number = 18;
    maxAge: number = 99;
    orderBy : string = 'lastActive';

    constructor(user: User) {
        this.gender = user.gender == 'male' ? 'female' : 'male';
    }
}

File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {

        public string Username { get; set; }
        public string Gender { get; set; }
        public int MaxAge { get; set; } = 150;
        public int MinAge { get; set; } = 18;
        public string OrderBy { get; set; } = "lastActive";
    }
}
